File: javascript/src/search_insert_position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
function searchInsert(nums, target) {
    let lo = 0;
    let hi = nums.length;
    while (lo <= hi) {
        const mid = Math.floor((lo + hi) / 2);
        if (nums[mid] < target) {
            lo = mid + 1;
        } else {
            hi = mid - 1;
        }
    }
    return lo;
}

export {searchInsert};

File: elixir/lib/search_insert_position.ex
defmodule SearchInsertPosition do
  @moduledoc """
  See: <https://leetcode.com/problems/search-insert-position/>
  """

  @spec search_insert(nums :: [integer], target :: integer) :: integer
  def search_insert(nums, target) do
    search_insert(nums, target, 0)
  end

  defp search_insert([], _, index), do: index

  defp search_insert([head | tail], target, index) do
    if head >= target do
      index
    else
      search_insert(tail, target, index + 1)
    end
  end
end
